app/battle.py

```diff
@@ -0,0 +1,26 @@
+from app.knight import Knight
+from typing import Dict
+
+
+def fight(knight1: Knight, knight2: Knight) -> None:
+    knight1.hp -= max(0, knight2.power - knight1.armour.protection)
+    knight2.hp -= max(0, knight1.power - knight2.armour.protection)
+    knight1.hp = max(0, knight1.hp)
+    knight2.hp = max(0, knight2.hp)
+
+
+def battle(knights_config: Dict[str, Dict]) -> Dict[str, int]:
+    lancelot = Knight(knights_config["lancelot"])
+    arthur = Knight(knights_config["arthur"])
+    mordred = Knight(knights_config["mordred"])
+    red_knight = Knight(knights_config["red_knight"])
```

Comment:
The creation of Knight instances is repetitive. You could simplify this by looping over the knight configurations and creating the instances dynamically. This would make your code more flexible and easier to maintain, especially if the number of knights were to increase in the future.